### ISSUE TITLE
Is is possible to provide helpers when the template is being rendered?

### ISSUE THREAD

**Issue by markdebhailis** [Handlebars-Net/Handlebars.Net#558]
Ask question related to the repo, e.g:
- How to achieve ... ?
- How does ... work?

Using this sample as a reference https://github.com/Handlebars-Net/Handlebars.Net#registering-helpers

I would like to pass the helper implementation to the template e.g. here's some psuedo code 

```
string source = @"Click here: {{link_to}}";

var template = Handlebars.Compile(source);

var data = new {
    url = "https://github.com/rexm/handlebars.net",
    text = "Handlebars.Net"
};

var helpers = new {
    link_to = (writer, context, parameters) => 
       {
           writer.WriteSafeString($"<a href='{context["url"]}'>{context["text"]}</a>");
      })
};

var result = template(data, helpers);
```

Is there a way to do this?

**Comment by oformaniuk** [comment]
No, it is not possible at the moment. I'd be happy to see PR enabling such behavior.

**Comment by rexm** [comment]
Helpers must be registered on the `IHandlebars` instance (or the static `Handlebars` class) before or after compilation — but not injected at render time. If you need different helpers per render, create separate `IHandlebars` instances with `Handlebars.Create()` and register the appropriate helpers on each. Closing as this is by design.
